cosmwasm/packages/wasmi-runtime/src/exports.rs

```diff
@@ -74,17 +94,26 @@ pub unsafe extern "C" fn ecall_init(
     let env = std::slice::from_raw_parts(env, env_len);
     let msg = std::slice::from_raw_parts(msg, msg_len);
     let result = panic::catch_unwind(|| {
-        let used_gas_ref = &mut *used_gas;
-        let result = crate::wasm::init(context, gas_limit, used_gas_ref, contract, env, msg);
+        let mut local_used_gas = *used_gas;
+        let result = crate::wasm::init(context, gas_limit, &mut local_used_gas, contract, env, msg);
+        *used_gas = local_used_gas;
         result_init_success_to_initresult(result)
     });
     if let Ok(res) = result {
         res
     } else {
-        error!("Call ecall_init panicked unexpectedly!");
-        // The enclave panicked. we do not report gas used in this case. (it should be initialized to 0)
-        InitResult::Failure {
-            err: EnclaveError::Panic,
+        *used_gas = gas_limit / 2;
```

Comment:
Can you add a comment here why we are just doing / 2? Also, is this something we may want to tune? If so make it a constant in consts.rs.

Repeat for all the other copies of this line

Reply:
Yeah. I wanted to consume a lot of gas for panics, but consuming all of it masks the real panic with an out of gas error.